packages/sdks/src/functions/evaluate/edge-runtime/edge-runtime.ts

```diff
@@ -1,32 +1,87 @@
+import { flattenState, getFunctionArguments } from '../helpers.js';
 import Interpreter from './acorn-interpreter.js';
 
+/**
+ * https://stackoverflow.com/a/46503625
+ */
+function patchInterpreter() {
+  const originalGetProperty = Interpreter.prototype.getProperty;
+  const originalSetProperty = Interpreter.prototype.setProperty;
+
+  function newGetProperty(this: typeof Interpreter, obj: any, name: any) {
+    if (obj == null || !obj._connected) {
+      return originalGetProperty.call(this, obj, name);
+    }
+
+    const value = obj._connected[name];
+    if (Array.isArray(value)) {
+      return this.nativeToPseudo(value);
+    }
+
+    if (typeof value === 'object') {
+      // if the value is an object itself, create another connected object
+      return this.createConnectedObject(value);
+    }
+    return value;
+  }
+  function newSetProperty(
+    this: typeof Interpreter,
+    obj: any,
+    name: any,
+    value: any,
+    opt_descriptor: any
+  ) {
+    if (obj == null || !obj._connected) {
+      return originalSetProperty.call(this, obj, name, value, opt_descriptor);
+    }
+
+    obj._connected[name] = this.pseudoToNative(value);
+  }
+
+  const getKeys: string[] = [];
+  const setKeys: string[] = [];
+  for (const key of Object.keys(Interpreter.prototype)) {
+    if (Interpreter.prototype[key] === originalGetProperty) {
+      getKeys.push(key);
+    }
+    if (Interpreter.prototype[key] === originalSetProperty) {
+      setKeys.push(key);
+    }
+  }
+
+  for (const key of getKeys) {
+    Interpreter.prototype[key] = newGetProperty;
+  }
+  for (const key of setKeys) {
+    Interpreter.prototype[key] = newSetProperty;
+  }
+
+  Interpreter.prototype.createConnectedObject = function (obj: any) {
+    const connectedObject = this.createObject(this.OBJECT);
+    connectedObject._connected = obj;
+    return connectedObject;
+  };
+}
+patchInterpreter();
+
 const processCode = (code: string) => {
-  return code
-    .split('\n')
-    .map((line) => {
-      const trimmed = line.trim();
-
-      // this async wrapper doesn't work in JS-interpreter, so we drop it.
-      if (line.includes('__awaiter')) return undefined;
-
-      // we find all state setter expressions and append a call to setRootState afterwards
-      const isStateSetter = trimmed.startsWith('state.');
-      if (!isStateSetter) return line;
-      const [lhs, rhs] = trimmed.split('=');
-      const setStr = lhs.replace('state.', '').trim();
-      const setExpr = `setRootState('${setStr}', ${rhs.trim()})`;
-      return `
-  ${line}
-  ${setExpr}
-  `;
-    })
-    .filter(Boolean)
-    .join('\n');
+  return (
+    code
+      // strip async/await polyfill: remove anything before 'function main()'
+      .replace(
+        /^.*?function main\(\)/,
+        `
+var __awaiter = function (e, t, n, r) {return r()},
+__generator = function (e, t) { return t() };
+function main()`
+      )
+      // replace ?. with .
+      .replace(/\?\./g, '.')
+  );
```

Comment:
Our `jsCode` blocks add a huge polyfill for `async/await` which reuses Promises. We need to remove that completely and replace it with identify functions because the js-interpreter does not support Promises